test/decimal_test.exs

```diff
@@ -9,6 +9,70 @@ defmodule DecimalTest do
 
   doctest Decimal
 
+  test "parse/1" do
+    assert Decimal.parse("123") == {d(1, 123, 0), ""}
+    assert Decimal.parse("+123") == {d(1, 123, 0), ""}
+    assert Decimal.parse("-123") == {d(-1, 123, 0), ""}
+    assert Decimal.parse("-123x") == {d(-1, 123, 0), "x"}
+    assert Decimal.parse("-123X") == {d(-1, 123, 0), "X"}
+
+    assert Decimal.parse("123.0") == {d(1, 1230, -1), ""}
+    assert Decimal.parse("+123.0") == {d(1, 1230, -1), ""}
+    assert Decimal.parse("-123.0") == {d(-1, 1230, -1), ""}
+    assert Decimal.parse("-123.0x") == {d(-1, 1230, -1), "x"}
+
+    assert Decimal.parse("1.5") == {d(1, 15, -1), ""}
+    assert Decimal.parse("+1.5") == {d(1, 15, -1), ""}
+    assert Decimal.parse("-1.5") == {d(-1, 15, -1), ""}
+    assert Decimal.parse("-1.5x") == {d(-1, 15, -1), "x"}
+
+    assert Decimal.parse("0") == {d(1, 0, 0), ""}
+    assert Decimal.parse("+0") == {d(1, 0, 0), ""}
+    assert Decimal.parse("-0") == {d(-1, 0, 0), ""}
+
+    assert Decimal.parse("0.0") == {d(1, 0, -1), ""}
+    assert Decimal.parse("-0.0") == {d(-1, 0, -1), ""}
+    assert Decimal.parse("+0.0") == {d(1, 0, -1), ""}
+
+    assert Decimal.parse("0.0.0") == {d(1, 0, -1), ".0"}
+    assert Decimal.parse("-0.0.0") == {d(-1, 0, -1), ".0"}
+    assert Decimal.parse("+0.0.0") == {d(1, 0, -1), ".0"}
+
+    assert Decimal.parse("0.") == :error
+    assert Decimal.parse(".0") == :error
```

Comment:
both of these values used to be allowed, not sure if we should keep them valid.

Reply:
They should be allowed according to the spec: http://speleotrove.com/decimal/daconvs.html. What made them break?

Reply:
Okay, I'll bring this behaviour back. The breakage was due to trying to simplifying parsing code.

Reply:
updated!